uk_election_timetables/elections/referendum.py

```
@@ -0,0 +1,9 @@
+from datetime import date
+
+from uk_election_timetables.election import Election
+
+
+class Referendum(Election):
+    @property
+    def sopn_publish_date(self) -> date:
+        raise NotImplementedError("No SOPN date for referenda")
```

I considered a couple of different approaches here:

Option 1:

```py
@property
def sopn_publish_date(self) -> Optional[date]:
    return None
```

Option 2:

```py
@property
def sopn_publish_date(self) -> date:
    raise NotImplementedError()
```

I decided on option 2. I think an exception will more actively break client code that doesn't explicitly handle it, but at least we get a notification in sentry saying it broke. `None` might fail loudly, or maybe we'll just silently render

> The list of candidates for this election will be published after [blank].

and we won't notice.

---

Surprisingly, there are not actually that many places where we need to handle this

I'm working on the EC site over in https://github.com/DemocracyClub/ec-postcode-lookup-pages/pull/186

We also probably need to handle it in:

- https://github.com/DemocracyClub/womblr/blob/803b3d95b7dbc2d958b2c8b09099f246bb1787b7/scraper.py#L161-L165
- https://github.com/DemocracyClub/WhoCanIVoteFor/blob/2217e23f535e562eff964243f0ba9ec0f4123b09/wcivf/apps/elections/models.py#L469

Think that is about it tbh

---

```
Errors should never pass silently.
Unless explicitly silenced.
```
:upside_down_face: 

Yeah, agree that an exception is better than `None` here